common/types.go

```diff
@@ -186,6 +186,42 @@ func (h Hash) Value() (driver.Value, error) {
 	return h[:], nil
 }
 
+// GetLastNumBitsFromHash returns the last num bits of the hash as int64
+func (h Hash) GetLastNumBitsFromHash(num int) int64 {
```

Comment:
Why return i64? The result can never be negative

Reply:
changed to uint64